doc/RichTextArea/RichTextArea.md

```diff
@@ -0,0 +1,742 @@
+
+This example illustrates how to style a read-only document created with the SimpleViewOnlyStyledModel and an application stylesheet:
+
+```java
+        SimpleViewOnlyStyledModel m = new SimpleViewOnlyStyledModel();
+        // add text segment using CSS style name (requires a style sheet)
+        m.addSegment("RichTextArea ", null, "HEADER");
+        // add text segment using direct style
+        m.addSegment("Demo", "-fx-font-size:200%;", null);
+        // newline
+        m.nl();
+```
+
+The default model for RichTextArea, the EditableRichTextModel, uses a number of style attributes (found in StyleAttrs class).  These attributes are applicable either to the whole paragraph (BACKGROUND, BULLET, FIRST_LINE_INDENT, ...) or to the individual text segments (BOLD, FONT_FAMILY, etc.).
+
+
+
+#### Export And Import
+
+StyledTextModel provides a common mechanism for importing/exporting styled text into/from the model via the following methods:
+
+- void **export**(TextPos start, TextPos end, StyledOutput out)
+- TextPos **replace**(StyleResolver, TextPos start, TextPos end, String text, boolean allowUndo)
+- TextPos **replace**(StyleResolver, TextPos start, TextPos end, StyledInput in, boolean allowUndo)
+
+The I/O interfaces **StyledInput** and **StyledOutput** provide the transport mechanism based on streaming of individual **StyledSegment**s.
+
+
+
+#### Clipboard
+
+The StyledTextModel provides capability to copy to and paste from the system clipboard in a variety of formats.  Several implementations are provided, all extending the **DataFormatHandler** class:
+
+|Format                             |Implementation                 |Export |Import |
+|:----------------------------------|:------------------------------|:------|:------|
+|DataFormat.HTML                    |**HtmlExportFormatHandler**    |yes    |no
+|DataFormat.PLAIN_TEXT              |**PlainTextFormatHandler**     |yes    |yes
+|RichTextFormatHandler.DATA_FORMAT  |**RichTextFormatHandler**      |yes    |yes
+|DataFormat.RTF                     |**RtfFormatHandler**           |yes    |yes (limited)
+
+The default model, EditableRichTextModel, copies plain text, HTML, RTF, as well as an internal format.  It also supports pasting plain text, internally formatted data, and a limited support for RTF.
+
+The following methods in StyledTextModel allow to register, remove, or query the data formats supported by the model:
+
+- Set<StyleAttribute<?>> **getSupportedAttributes**()
+- void **registerDataFormatHandler**(DataFormatHandler, boolean forExport, boolean forImport, int priority)
+- void **removeDataFormatHandler**(DataFormat, boolean forExport)
+  
+At the control level, the following methods allow for data transfer in any format supported by the model:
+
+- void **read**(DataFormat)
+- void **read**(DataFormat, InputStream)
+- void **write**(OutputStream)
+- void **write**(DataFormat, OutputStream)
+
+
+#### Adding New Attributes
```

Comment:
This logically belongs with the extensibility section.

Reply:
... and it is (enable outline view to see)